tests/pql/test_query.py

```diff
@@ -7,3 +9,175 @@ def test_simple_evaluator_query():
+def test_not_valid_such_statement():
+    with pytest.raises(ValueError, match="Expected token 'such', got 'sum'"):
+        evaluate_query(
+            """ stmt s1;
+                Select s1 sum that Parent(s1, "x")
+           """
+        )
+
+
+def test_not_valid_that_statement():
+    with pytest.raises(ValueError, match="Expected token 'that', got 'dat'"):
+        evaluate_query(
+            """ stmt s1;
+                Select s1 such dat Parent(s1, "x")
+           """
+        )
+
+
+def test_relation_parent_in_query():
+    result = evaluate_query(
+        """ stmt s1;
+            Select s1 such that Parent(s1, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Parent"
+
+
+def test_relation_parent_star_in_query():
+    result = evaluate_query(
+        """ stmt s1;
+            Select s1 such that Parent*(s1, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Parent*"
+
+
+def test_relation_follows_in_query():
+    result = evaluate_query(
+        """ assign a2;
+            Select a2 such that Follows(a2, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Follows"
+
+
+def test_relation_follows_star_in_query():
+    result = evaluate_query(
+        """ assign a2;
+            Select a2 such that Follows*(a2, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Follows*"
+
+
+def test_relation_modifies_in_query():
+    result = evaluate_query(
+        """ assign a2;
+            Select a2 such that Modifies(a2, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Modifies"
+
+
+def test_relation_uses_in_query():
+    result = evaluate_query(
+        """ while w3;
+            Select w3 such that Uses(w3, "x")
+           """
+    )
+
+    assert result[0]["relation"] == "Uses"
+
+
+def test_valid_parameters_with_string_in_relations():
+    result = evaluate_query(
+        """ while w3;
+            Select w3 such that Uses(w3, "x")
+           """
+    )
+
+    assert result[0]["parameters"][0] == "w3"
+    assert result[0]["parameters"][1] == '"x"'
+
+
+def test_valid_parameters_with_integer_in_relations():
+    result = evaluate_query(
+        """ while w3;
+            Select w3 such that Uses(20, w3)
+           """
+    )
+
+    assert result[0]["parameters"][0] == "20"
+    assert result[0]["parameters"][1] == "w3"
+
+
+def test_not_valid_parameters_in_relations():
+    with pytest.raises(ValueError, match="Variable 'xd1' is not declared"):
+        evaluate_query(
+            """
+            while w3;
+            Select w3 such that Uses(xd1, w3)
+            """
+        )
+
+
+def test_complex_query_evaluator():
+    evaluate_query(
+        """
+            while w3;
+            Select w3 such that Uses(20, w3) with w3.atrrName = "x"
+            """
+    )
+
+
+def test_():
+    result = evaluate_query(
+        """
+            while w3;
+            Select w3 such that Uses(20, w3) with w3.atrrName = "x"
+        """
+    )
+
```

Comment:
Do testow z withem pododawaj to:
```suggestion

assert len(result[0]["with"]) == 1
```